google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/BitReversedSequenceStyleGenerator.java

```diff
@@ -157,8 +187,59 @@ protected DatabaseStructure buildTableStructure(
         type.getReturnedClass());
   }
 
+  @Override
+  public void configure(Type type, Properties params, ServiceRegistry serviceRegistry)
+      throws MappingException {
+    super.configure(type, params, serviceRegistry);
+    configureExcludedRanges(this.sequenceName.getObjectName().getText(), params);
+  }
+
+  @VisibleForTesting
+  void configureExcludedRanges(String sequenceName, Properties params) throws MappingException {
+    this.excludeRanges = parseExcludedRanges(sequenceName, params);
+  }
+
   @VisibleForTesting
-  static final int MAX_ATTEMPTS = 100;
+  static List<Range<Long>> parseExcludedRanges(String sequenceName, Properties params) {
```

Comment:
Nit: Can we have this method arranged after all the public methods of the class?

Reply:
I would prefer not to do that. These methods form a logical block of code together with the `public configure(..)` method. The main reason that it consists of a couple of package-private methods together with the public `configure` method is to make it more testable. Keeping methods that form a single block of logic together improves the readability of that block, as you don't have to jump back and forth a lot in the source code.

Note that the [Java style guide](https://google.github.io/styleguide/javaguide.html#s3.4.2-ordering-class-contents) requires the source code to use some form of logical ordering, and has some strict rules for specific content such as constructors and overloads, but not for the order of `public` and `private` methods.